internal/services/computefleet/compute_fleet_resource.go

```diff
@@ -0,0 +1,1006 @@
+
+		"compute_api_version": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			Computed:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"identity": commonschema.SystemAssignedUserAssignedIdentityOptional(),
+
+		"regular_priority_profile": {
+			Type:         pluginsdk.TypeList,
+			Optional:     true,
+			MaxItems:     1,
+			AtLeastOneOf: []string{"regular_priority_profile", "spot_priority_profile"},
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"capacity": {
+						Type:         pluginsdk.TypeInt,
+						Required:     true,
+						ValidateFunc: validation.IntBetween(1, 10000),
+					},
+
+					"allocation_strategy": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ForceNew:     true,
+						Default:      string(fleets.RegularPriorityAllocationStrategyLowestPrice),
+						ValidateFunc: validation.StringInSlice(fleets.PossibleValuesForRegularPriorityAllocationStrategy(), false),
+					},
+
+					"min_capacity": {
+						Type:         pluginsdk.TypeInt,
+						ForceNew:     true,
+						Optional:     true,
+						Default:      0,
+						ValidateFunc: validation.IntAtLeast(0),
+					},
+				},
+			},
+		},
+
+		"spot_priority_profile": {
+			Type:         pluginsdk.TypeList,
+			Optional:     true,
+			MaxItems:     1,
+			AtLeastOneOf: []string{"regular_priority_profile", "spot_priority_profile"},
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"capacity": {
+						Type:         pluginsdk.TypeInt,
+						Required:     true,
+						ValidateFunc: validation.IntBetween(1, 10000),
+					},
+
+					"allocation_strategy": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ForceNew:     true,
+						Default:      string(fleets.SpotAllocationStrategyPriceCapacityOptimized),
+						ValidateFunc: validation.StringInSlice(fleets.PossibleValuesForSpotAllocationStrategy(), false),
+					},
+
+					"eviction_policy": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ForceNew:     true,
+						Default:      string(fleets.EvictionPolicyDelete),
+						ValidateFunc: validation.StringInSlice(fleets.PossibleValuesForEvictionPolicy(), false),
+					},
+
+					"maintain_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Optional: true,
+						ForceNew: true,
+						Default:  true,
+					},
+
+					"max_hourly_price_per_vm": {
+						Type:         pluginsdk.TypeFloat,
+						Optional:     true,
+						ForceNew:     true,
+						Default:      -1,
+						ValidateFunc: computeValidate.SpotMaxPrice,
+					},
+
+					"min_capacity": {
+						Type:         pluginsdk.TypeInt,
+						Optional:     true,
+						ForceNew:     true,
+						Default:      0,
+						ValidateFunc: validation.IntAtLeast(0),
+					},
+				},
+			},
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func virtualMachineAdditionalCapabilitiesSchema() *pluginsdk.Schema {
```

Comment:
Could this schema be moved into the rest of the explicit definition? It seems this function is only called once.